### PR TITLE
Insert explicit cast for cython 0.25.2

### LINKED ISSUE #1983: Cannot build Chainer with Cython 0.25.2
When I build Chainer with Cython 0.25.2, it was crashed.

### PR DESCRIPTION
**Pull request by unnonouno**
I inserted explicit casts to avoid cython 0.25.2 problem.
fix #1983

### CODE REVIEW

File: cupy/core/elementwise.pxi [hunk 1]
```diff
@@ -185,13 +185,13 @@ cpdef tuple _reduce_dims(list args, tuple params, tuple shape):
     if cnt == ndim:
         return args, shape
     if cnt == 1:
-        newshape.assign(1, vecshape[axis])
+        newshape.assign(1, <long>vecshape[axis])
```
Comment (gwtnb) [inline cupy/core/elementwise.pxi]: Py_ssize_t

File: cupy/core/elementwise.pxi [hunk 2]
```diff
@@ -185,13 +185,13 @@ cpdef tuple _reduce_dims(list args, tuple params, tuple shape):
...
         ret = []
         for i, a in enumerate(args):
             if is_array_flags[i]:
                 arr = a
                 arr = arr.view()
-                newstrides.assign(1, arr._strides[axis])
+                newstrides.assign(1, <long>arr._strides[axis])
```
Comment (gwtnb) [inline cupy/core/elementwise.pxi]: Py_ssize_t

File: cupy/core/core.pyx [hunk 3]
```diff
@@ -492,8 +492,8 @@ cdef class ndarray:
             newarray = ndarray(self.shape, self.dtype)
             elementwise_copy(self, newarray)
 
-        newarray._shape.assign(1, self.size)
-        newarray._strides.assign(1, self.itemsize)
+        newarray._shape.assign(1, <long>self.size)
```
Comment (gwtnb) [inline cupy/core/core.pyx]: Py_ssize_t

File: cupy/core/core.pyx [hunk 4]
```diff
@@ -492,8 +492,8 @@ cdef class ndarray:
...
+        newarray._strides.assign(1, <long>self.itemsize)
```
Comment (gwtnb) [inline cupy/core/core.pyx]: Py_ssize_t

File: cupy/core/core.pyx (hunk 3 above)
Comment (unnonouno) [inline cupy/core/core.pyx]: It causes an error. _shape is long.

Comment (gwtnb) [inline cupy/core/core.pyx]: oh

**Comment by gwtnb** [comment]
Ref: https://github.com/cython/cython/issues/1549

**Comment by gwtnb** [comment]
It may be problematic to use `long` because `sizeof(long) == 4` on Windows.

**Comment by unnonouno** [comment]
I don't know why `Py_ssize_t` causes an error. `ndarray._shape` looks a vector of `Py_ssize_t`.

**Comment by unnonouno** [comment]
I found a workaround!

**Comment by unnonouno** [comment]
- `std::vector::assign` has two interfaces. One is for iterators and is a template method, and the other is for size_t
- This method checks interface of template methods, and found type inconsistency here (and cython faces a bug). Actually this `assign` is not the first one but the second one. So this inconsistency will be ignored.
- I fixed inconsistency. I mean, I inserted casts to make both arguments have the same types

**Pull request merged**